src/main/java/io/confluent/idesidecar/restapi/models/graph/RealDirectFetcher.java

```diff
@@ -78,6 +78,21 @@ public List<DirectConnection> getConnections() {
         .toList();
   }
 
+  @Override
+  public DirectConnection getDirectConnectionByID(String connectionID) {
```

Comment:
I feel this function should throw an error if a connection with the given ID exists but is not of type `DIRECT`, so that it's aligned with the behavior of the API for CCloud connections. We should cover this case in a test, too.

Reply:
agreed--- made the changes